backend/src/main/java/saci/application/model/ScoreResponse.java

```diff
@@ -0,0 +1,87 @@
+package saci.application.model;
+
+import java.util.Optional;
+import saci.domain.model.Level;
+
+public class ScoreResponse {
+    private Optional<LevelResponse> currentLevel;
+    private Optional<LevelResponse> nextLevel;
+
+    private ScoreResponse(Builder builder) {
+        this.currentLevel = builder.currentLevel;
+        this.nextLevel = builder.nextLevel;
+    }
+
+    public Optional<LevelResponse> getCurrentLevel() {
+        return currentLevel;
+    }
+
+    public Optional<LevelResponse> getNextLevel() {
+        return nextLevel;
+    }
+
+    public static class Builder {
+        private Optional<LevelResponse> currentLevel = Optional.empty();
+        private Optional<LevelResponse> nextLevel = Optional.empty();
+
+        public Builder currentLevel(Optional<Level> currentLevel) {
+            this.currentLevel =
+                    currentLevel.map(
+                            level -> {
+                                LevelResponse levelResponse = new LevelResponse();
```

Comment:
Also add a builder for every "new Object()" instantiation that you hav ein your classes.